packages/compass-connections/src/components/connection-list/connection-list.tsx

```diff
@@ -171,6 +172,9 @@ function ConnectionList({
           {favoriteConnections.map((connectionInfo, index) => (
             <li
               data-testid="favorite-connection"
+              data-id={`favorite-connection-${getConnectionTitle(
```

Comment:
This function is not meant for this use case, makes sense in places where we don't know if a connection is a favorite or not, but using it here is a bit unexpected. Since this connection is guaranteed to be a favorite why not something like `connectionInfo?.favorite?.name` instead?